dashboard/app/assets/javascripts/components/report_abuse_form.jsx

```diff
@@ -0,0 +1,139 @@
+window.dashboard = window.dashboard || {};
+
+/**
+ * A component containing some text/links for projects that have had abuse
+ * reported. This is used in our blocking AbuseBox, in the share dialog, and
+ * in our smaller alert in apps.
+ */
+window.dashboard.ReportAbuseForm = (function (React) {
+  var INPUT_WIDTH = 500;
+  // dropdown width is wider so that it still lines up with inputs (which have
+  // padding)
+  var DROPDOWN_WIDTH = 514;
+
+  /**
+   * A dropdown with the set of ages we use across our site (4-20, 21+)
+   */
+  var AgeDropdown = React.createClass({
+    propTypes: {
+      age: React.PropTypes.string
+    },
+
+    render: function () {
+      var style = $.extend({}, {width: DROPDOWN_WIDTH}, this.props.style);
+
+      var ages = ['', '4', '5', '6', '7', '8', '9', '10', '11', '12', '13', '14',
+        '15', '16', '17', '18', '19', '20', '21+'];
+
+      if (this.props.age !== null && ages.indexOf(this.props.age) == -1) {
+        throw new Error('Invalid age: ' + this.props.age);
+      }
+
+      return (
+        <select style={style} value={this.props.age}>{
+          ages.map(function (age) {
+            return <option key={age} value={age}>{age}</option>;
+          })
+        }</select>
+      );
+    }
+  });
+
+  return React.createClass({
+    propTypes: {
+      i18n: React.PropTypes.object.isRequired,
+      csrfToken: React.PropTypes.string.isRequired,
+      abuseUrl: React.PropTypes.string.isRequired,
+      name: React.PropTypes.string,
+      email: React.PropTypes.string,
+      age: React.PropTypes.string,
+    },
+
+    /**
+     * Extracts a channel id from the given abuse url
+     * @returns {string} Channel id, or undefined if we can't get one.
+     */
+    getChannelId: function () {
+      var match = /.*\/projects\/[^\/]+\/([^\/]+)/.exec(this.props.abuseUrl);
```

Comment:
No. What happens is you click "report abuse" and get taken to a new page/route where the project is not loaded. If we're worried about this regex, we could pass the channel id as a query param (i.e. go to /report_abuse?channel_id=foobar)